Source/Core/Core/IOS/Network/IP/Top.cpp

```diff
@@ -554,48 +556,52 @@ IPCCommandResult NetIPTop::HandleGetPeerNameRequest(const IOCtlRequest& request)
                       std::min<size_t>(sizeof(sa.sa_data), request.buffer_out_size - 2));
   }
 
-  INFO_LOG(IOS_NET, "IOCTL_SO_GETPEERNAME(%x)", fd);
+  INFO_LOG_FMT(IOS_NET, "IOCTL_SO_GETPEERNAME({:x})", fd);
   return GetDefaultReply(ret);
 }
 
 IPCCommandResult NetIPTop::HandleGetHostIDRequest(const IOCtlRequest& request)
 {
   const DefaultInterface interface = GetSystemDefaultInterfaceOrFallback();
   const u32 host_ip = Common::swap32(interface.inet);
-  INFO_LOG(IOS_NET, "IOCTL_SO_GETHOSTID = %u.%u.%u.%u", host_ip >> 24, (host_ip >> 16) & 0xFF,
-           (host_ip >> 8) & 0xFF, host_ip & 0xFF);
+  INFO_LOG_FMT(IOS_NET, "IOCTL_SO_GETHOSTID = {}.{}.{}.{}", host_ip >> 24, (host_ip >> 16) & 0xFF,
+               (host_ip >> 8) & 0xFF, host_ip & 0xFF);
   return GetDefaultReply(host_ip);
 }
 
 IPCCommandResult NetIPTop::HandleInetAToNRequest(const IOCtlRequest& request)
 {
-  std::string hostname = Memory::GetString(request.buffer_in);
+  const std::string hostname = Memory::GetString(request.buffer_in);
   struct hostent* remoteHost = gethostbyname(hostname.c_str());
 
   if (remoteHost == nullptr || remoteHost->h_addr_list == nullptr ||
       remoteHost->h_addr_list[0] == nullptr)
   {
-    INFO_LOG(IOS_NET,
-             "IOCTL_SO_INETATON = -1 "
-             "%s, BufferIn: (%08x, %i), BufferOut: (%08x, %i), IP Found: None",
-             hostname.c_str(), request.buffer_in, request.buffer_in_size, request.buffer_out,
-             request.buffer_out_size);
+    INFO_LOG_FMT(IOS_NET,
+                 "IOCTL_SO_INETATON = -1 "
+                 "{}, BufferIn: ({:08x}, {}), BufferOut: ({:08x}, {}), IP Found: None",
+                 hostname, request.buffer_in, request.buffer_in_size, request.buffer_out,
+                 request.buffer_out_size);
+
     return GetDefaultReply(0);
   }
 
-  Memory::Write_U32(Common::swap32(*(u32*)remoteHost->h_addr_list[0]), request.buffer_out);
-  INFO_LOG(IOS_NET,
-           "IOCTL_SO_INETATON = 0 "
-           "%s, BufferIn: (%08x, %i), BufferOut: (%08x, %i), IP Found: %08X",
-           hostname.c_str(), request.buffer_in, request.buffer_in_size, request.buffer_out,
-           request.buffer_out_size, Common::swap32(*(u32*)remoteHost->h_addr_list[0]));
+  const auto ip = Common::swap32(*(u32*)remoteHost->h_addr_list[0]);
```

Comment:
You changed other places to C++ style casts, but not this one - was that on purpose or did you simply miss it?

Reply:
Changed it over. Also we can use the swap32 variant that takes a `u8*` to avoid invoking undefined behavior wrt pointer casting and dereferencing.